test/spec/core/localizer.js

```diff
@@ -0,0 +1,40 @@
+describe('iD.coreLocalizer', function () {
+    describe('#floatParser', function () {
+        it('roundtrips English numbers', function () {
```

Comment:
At first glance it look a bit like you took a shortcut here, by only checking that the stringify->parse chain results in correct results. Technically this wouldn't catch the error where all locales would fall back to English. Is this on purpose to make tests pass on test environments which don't support `Intl`?

Reply:
Unfortunately, none of the Node.js configurations on CI have been built with `Intl` support. 😕 This test can only test roundtripping in the most basic sense, which I guess does cover some of the older browser versions supported by iD.

Reply:
ok, I see. Do you know if #8764 (switch from phantomjs to karma) would change anything in this regard?

Reply:
I wouldn't expect it to – this is about Node.js itself rather than the test harness – but I pushed openstreetmap/iD@bd67bc3f5b964f314ac44fb417df8df942606531 so you can try it out.

Reply:
karma seems to be capable of executing tests in actual browsers. In PR #8764 it is configured to run in a (headless version of) Chrome; see https://github.com/openstreetmap/iD/pull/8764/files#diff-25bdb46a5428d318648ec9eb46a99c0268d8c96b672c46e2247a685bc3e384fbR79-R83.

I just tried it out and it passes the `formats Bengali numbers` test merged in from `bd67bc3`. However, there seems to be an issue with the Arabic numbers test, where I get the following error:

```
Chrome Headless 88.0.4324.27 (Linux x86_64) iD.coreLocalizer #floatParser roundtrips Arabic numbers FAILED
	AssertionError: expected NaN to deeply equal -0.1
	    at Context.<anonymous> (test/spec/core/localizer.js:25:54)
```

All other tests work and pass just fine.

Reply:
ok, I could confirm the issue also on a current "regular" Chromium browser (v95), as well as with Firefox (v94). The issue seems to be that the `floatParser` isn't able to handle input of negative numbers with any right-to-left script:

```javascript
floatParser('ar-EG')((-0.1).toLocaleString('ar-EG'))
# NaN
floatParser('he')((-0.1).toLocaleString('he'))
# NaN
```

In these cases, the result from `toLocaleString` include either a Unicode [left-to-right mark (LRM, U+200E)](https://www.fileformat.info/info/unicode/char/200e/index.htm) or a [arabic letter mark (ALM, U+061C)](https://www.fileformat.info/info/unicode/char/061c/index.htm) as the first character. Stripping these from the input string in the parser function solves the issue, but I don't know if there are any potential side effects this could have. :thinking: 

```javascript
    localizer.floatParser = (locale) => {
        // […]
        return (string) => {
            string = string.trim().replace(/^[\u200E\u200F\u061C]/, '').replace(group, '')
                .replace(decimal, '.')
                .replace(numeral, getIndex);
            return string ? +string : NaN;
        };
    };
```

Reply:
> In these cases, the result from `toLocaleString` include either a Unicode [left-to-right mark (LRM, U+200E)](https://www.fileformat.info/info/unicode/char/200e/index.htm) or a [arabic letter mark (ALM, U+061C)](https://www.fileformat.info/info/unicode/char/061c/index.htm) as the first character.

I guess that makes sense: the Unicode Character Database gives the hyphen-minus character `-` a bidi character type of European Number Separator, which is weakly directional. The number formatter uses the ALM in Arabic and LRM in Hebrew to ensure the hyphen-minus appears on the correct side of the digits, e.g., “-3.14159” and “؜-٣٫١٤١٥٩”, and not as “‏-3.14159” and “-٣٫١٤١٥٩”.

> Stripping these from the input string in the parser function solves the issue, but I don't know if there are any potential side effects this could have. :thinking:

`Number.prototype.formatToParts` treats these bidi characters as `literal`s, so we can remove them rather generically. I don’t think this would be any more of a problem than removing the grouping symbols, since the characters are there purely to ensure correct layout; we’re parsing the characters in logical order rather than visual order.

Reply:
> `Number.prototype.formatToParts` treats these bidi characters as `literal`s, so we can remove them rather generically.

I added code to strip out any `literal`s, including these bidirectional characters in the locales that use them.